### PR TITLE
Handling of procedure locals allocations and deallocations

### PR DESCRIPTION
**Pull request by bobbinth**
This PR adds handling of allocations and deallocations for procedure locals in the new Assembler structure.

### CODE REVIEW

File: assembly/src/todo/span_builder.rs [hunk 1]
```diff
@@ -9,14 +9,40 @@ use super::{AssemblerError, Borrow, CodeBlock, DecoratorList, Operation, Vec};
...
 impl SpanBuilder {
+    // CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+    /// Returns a new [SpanBuilder] instantiated with the specified optional wrapper.
+    ///
+    /// If the wrapper is provided, the prologue of the wrapper is immediately appended to the
+    /// vector of span operations. The epilogue of the wrapper is appended to the list of
+    /// operations upon consumption of the builder via `extract_final_span_into()` method.
+    pub(super) fn new(wrapper: Option<BodyWrapper>) -> Self {
+        if let Some(wrapper) = wrapper {
```
Comment (vlopes11) [inline assembly/src/todo/span_builder.rs:24]: minor nit: this looks like a

```rust
match wrapper {
    Some(wrapper) => (..),
    None => Self::default(),
}
```

**Review by vlopes11 — APPROVED** [review]
looks good!

not a blocker: the only point to think about is if the prologue/epilogue of the span builder isn't going to be an exclusive construction of the procedure (to define the number of locals). If yes, it might be worthy to make a dedicated `extract_span_into_with_prologue` instead of changing the structure.

File: assembly/src/todo/span_builder.rs (hunk 1 above)
Comment (bobbinth) [reply assembly/src/todo/span_builder.rs]: Thanks! Fixed!

**Comment by bobbinth** [comment]
> the only point to think about is if the prologue/epilogue of the span builder isn't going to be an exclusive construction of the procedure (to define the number of locals).

I think we can keep it more general for now. There might be other instances when we want to have wrappers around code blocks - though, I can't come up with any specific ones at the moment.

**Pull request merged**